tests/helpers/utils.py

```diff
@@ -140,12 +140,18 @@ def normalizer_data_from_string(s):
     # so we use inspect.cleandoc to remove leading indentation
     s = inspect.cleandoc(s)
 
-    category_mappings = defaultdict(list)
-    synset_mappings = defaultdict(list)
+    category_mappings = {}
+    synset_mappings = {}
     for line in s.splitlines():
         tokens = line.split(" ")
 
         curie = tokens[0]
+        if curie not in category_mappings:
+            category_mappings[curie] = []
+        if curie not in synset_mappings:
+            # Curies are always self-synonyms
+            synset_mappings[curie] = [curie]
```

Comment:
Is line 160 now unreachable? Why do this instead of `defaultdict`? What happens if `curie` _is_ in `synset_mappings`? Is that possible, and would it then not be its own synonym?

Reply:
Our code relies on the normalizer returning the given curie in the list of equivalent_identifiers. I confirmed that this is the case in the real node normalizer:

<img width="1432" alt="Screen Shot 2021-04-07 at 1 33 33 PM" src="https://user-images.githubusercontent.com/1634866/113909929-3e44e580-97a6-11eb-9ab0-a5401ba2934d.png">

This was the bug that caused test_issue_102 to fail. If the curie is not listed in equivalent_identifiers we never do any synonymizing. You're correct that we can use a defaultdict to streamline this and that line 160 is unreachable.